client/src/pages/ContestDetails/ContestDetails.tsx

```diff
@@ -15,18 +15,27 @@ import { useSocket } from '../../context/useSocketContext';
 import useStyles from './useStyles';
 import demoProfilePhoto from '../../Images/demo-profile-photo.png';
 import sampleContestData from './SampleContestData';
+import { getAllSubmissions } from '../../helpers/APICalls/submission';
+import { Submission } from '../../interface/Submission';
 
-export default function ContestDetails(): JSX.Element {
+export default function ContestDetails({ match }: RouteComponentProps): JSX.Element {
   const classes = useStyles();
+  const [submissionObj, setSubmissionObj] = useState<Submission[]>(Object);
+  const [contestId, setContestId] = useState<string>('');
   const { loggedInUser } = useAuth();
   const { initSocket } = useSocket();
 
   const history = useHistory();
-  const { title, description, prizeAmount, user, submissions } = sampleContestData();
+  const { title, description, prizeAmount, user } = sampleContestData();
 
   useEffect(() => {
     initSocket();
-  }, [initSocket]);
+    const params = match.params as { id: string };
+    setContestId(params.id);
+    getAllSubmissions(params.id).then((data) => {
+      setSubmissionObj(data.submission);
+    });
+  }, [initSocket, match]);
```

Comment:
We don't need to initialize the socket within this useEffect, you can exclude the function as well as the dependency

Reply:
removed